stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -300,6 +302,69 @@ internal class StripePaymentController internal constructor(
         stripeRepository.retrieveSource(sourceId, clientSecret, requestOptions, callback)
     }
 
+    override fun authenticateAlipay(
+        intent: StripeIntent,
+        clientSecret: String,
+        stripeAccountId: String?,
+        authenticationHandler: AlipayAuthenticationHandler,
+        callback: ApiResultCallback<PaymentIntentResult>
+    ) {
+        AlipayAuthenticationTask(
+            intent,
+            authenticationHandler,
+            object : ApiResultCallback<Int> {
+                override fun onSuccess(result: Int) {
+                    val requestOptions = ApiRequest.Options(
+                        apiKey = publishableKey,
+                        stripeAccount = stripeAccountId
+                    )
+
+                    stripeRepository.retrieveIntent(
+                        clientSecret,
```

Comment:
@mshafrir-stripe  Can we use `intent.clientSecret` here instead? That would simplify the interface instead of passing the secret around everywhere. Unfortunately, this field is nullable. When would it ever be null?

Reply:
Seems fine to do that. The `clientSecret` field would only be null in the case of an error. Maybe `intent.clientSecret.orEmpty()`?